lib/Conversion/TorchToTosa/TorchToTosa.cpp

```diff
@@ -1892,26 +1919,184 @@ LogicalResult ConvertAtenOp<AtenBatchNormOp>::matchAndRewrite(
+
+  auto inputType = adaptor.input().getType().cast<RankedTensorType>();
+  if (inputType.getRank() > 4)
+    return op.emitError("Only up to 4D tensors are supported");
+
+  auto outType = getTypeConverter()->convertType(op.getType(0));
+
+  // Note: cudnn_enabled is not handled.
+
+  // FIXME: Handle the None cases for the optional parameters.
+  if (adaptor.weight().getType().isa<Torch::NoneType>())
+    return op.emitError("Unsupported None for weight");
+  if (adaptor.bias().getType().isa<Torch::NoneType>())
+    return op.emitError("Unsupported None for bias");
+
+  auto weightType = adaptor.weight().getType().cast<RankedTensorType>();
+  auto biasType = adaptor.bias().getType().cast<RankedTensorType>();
+  int64_t inputRank = inputType.getRank();
+  Type elemTy = inputType.getElementType();
+
+  // Check if all the arguments meet the requirements.
+  SmallVector<int64_t> normalizedShapeSizesInt;
+  if (!matchPattern(op.normalized_shape(),
+                    m_TorchConstantIntList(normalizedShapeSizesInt))) {
+    return rewriter.notifyMatchFailure(op, "Unimplemented normalized_shape not"
+                                           "constructed from ListConstruct");
+  }
+  int64_t normalizedShapeRank = normalizedShapeSizesInt.size();
+  if (weightType.getRank() != normalizedShapeRank ||
+      biasType.getRank() != normalizedShapeRank ||
+      inputRank < normalizedShapeRank || normalizedShapeRank < 1)
+    return rewriter.notifyMatchFailure(op, "Input or weight or bias shape or"
+                                           "normalized shape not compatible");
+
+  // Check all the dimensions match the normalized_shape, only static shapes as
+  // of now
+  int64_t meanAndVarShapeRank = inputRank - normalizedShapeSizesInt.size();
+  for (auto en : llvm::enumerate((normalizedShapeSizesInt))) {
+    int64_t index = en.index();
+    int64_t value = en.value();
+    if (inputType.getShape()[index + meanAndVarShapeRank] != value ||
+        weightType.getShape()[index] != value ||
+        biasType.getShape()[index] != value)
+      return op.emitError("mismatching contracting dimension");
+  }
+
+  // Helper for computing mean and variance.
+  auto computeSumAndReshape = [&](Value toReduce, RankedTensorType toReduceType,
+                                  Type outType, SmallVector<int64_t> outShape) {
+    Value sumDiv = toReduce;
+    SmallVector<int64_t> toReduceShape(toReduceType.getShape().begin(),
+                                       toReduceType.getShape().end());
+    while (static_cast<int64_t>(toReduceShape.size()) != meanAndVarShapeRank) {
+      toReduceShape.back() = 1;
+      sumDiv =
+          rewriter
+              .create<tosa::ReduceSumOp>(
+                  op.getLoc(),
+                  RankedTensorType::get(toReduceShape,
+                                        inputType.getElementType()),
+                  sumDiv, rewriter.getI64IntegerAttr(toReduceShape.size() - 1))
+              .getResult();
```

Comment:
nit: all operations with `::mlir::OpTrait::OneResult` trait doesn't need `getResult()`. This is the same for reshape and a few other ops.